xgi/convert.py

```diff
@@ -134,6 +134,38 @@ def convert_to_graph(H):
     return G
 
 
+def convert_to_line_graph(H):
+    """Line graph of the hypergraph.
+    
+    The line graph of the hypergraph `H` is the graph whose
+    nodes correspond to each hyperedge in `H`, linked together
+    if they share at least one vertex.
+
+    Parameters
+    ----------
+    H : Hypergraph
+        The hypergraph of interest
+
+    Returns
+    -------
+    LG : networkx.Graph
+         The line graph associated to the Hypergraph
+    """
+
+    LG = nx.Graph()
+
+    edge_label_dict = {
+        tuple(edge): index for index, edge in enumerate(H.edges.members())
+    }
+    LG.add_nodes_from(edge_label_dict.values())
```

Comment:
Current code is correct but will add nodes in their order of appearances in hyperedges. The following adds them in an ordered way:
```suggestion
    nodes = sorted(set(edge_label_dict.values()))
    LG.add_nodes_from(nodes)
```